testing/docker-compose.yml

```
@@ -0,0 +1,98 @@
+version: "3.7"
+services:
+  s3_backend:
+    command: server /data
+    container_name: s3
+    environment:
+      - MINIO_ACCESS_KEY=ElixirID
+      - MINIO_SECRET_KEY=987654321
+    healthcheck:
+      test: ["CMD", "curl", "-fkq", "https://localhost:9000/minio/health/live"]
+      interval: 5s
+      timeout: 20s
+      retries: 3
+    image: minio/minio:RELEASE.2021-02-14T04-01-33Z
+    ports:
+      - "9000:9000"
+    volumes:
+      - data:/data
+  createbucket:
+    container_name: buckets
+    image: minio/mc
+    depends_on:
+      - s3_backend
+    entrypoint: >
+      /bin/sh -c "
+      /usr/bin/mc config host add s3 http://s3:9000 ElixirID 987654321;
+      /usr/bin/mc rm -r --force s3/test;
+      /usr/bin/mc mb s3/test;
+      /usr/bin/mc mb s3/download;
+      /usr/bin/mc policy set public s3/download;
+      exit 0;
+      "
+    volumes:
+      - ./certs/ca.crt:/etc/ssl/certs/public.crt
+  mq_server:
+    container_name: mq
+    image: rabbitmq:3.7.8-management-alpine
+    ports:
+      - "15672:15672"
+      - "5672:5672"
+      - "5671:5671"
+    volumes:
+      - ./defs.json:/etc/rabbitmq/defs.json
+      - ./rabbitmq.conf:/etc/rabbitmq/rabbitmq.conf
+      - ./certs/ca.crt:/etc/rabbitmq/ssl/ca.crt
+      - ./certs/mq.crt:/etc/rabbitmq/ssl/mq.crt
+      - ./certs/mq.key:/etc/rabbitmq/ssl/mq.key
+    healthcheck:
+      test: [ "CMD", "nc", "-z", "localhost", "5672" ]
```

```suggestion
      test: [ "CMD", "rabbitmq-diagnostics", "-q", "check_running" ]
```

It's better to use rabitMQs built in tests.
This can be expanded to run the following `rabbitmq-diagnostics -q check_running && rabbitmq-diagnostics -q check_local_alarms`